modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMStructureLocalServiceImpl.java

```diff
@@ -2289,6 +2350,29 @@ private void _validate(
 		}
 	}
 
+	private void _validateExternalReferenceCode(
+		String externalReferenceCode, long groupId, long classNameId) {
+
+		if (Validator.isNull(externalReferenceCode)) {
+			return;
+		}
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-34651")) {
+			throw new UnsupportedOperationException();
+		}
+
+		DDMStructure structure = ddmStructurePersistence.fetchByERC_G_C(
+			externalReferenceCode, groupId, classNameId);
+
+		if (structure != null) {
```

Comment:
We must consider when the user updates and sends the same ERC in the PUT request. 
The current code would throw an exception.